CLI: Implement `auth status` command

fredrikekelund opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "Fixes" keyword and use "Related to" instead.
-->

## Related issues

- Fixes STU-948

## Proposed Changes

This PR is based on @youknowriad's work in #1958. It implements a `studio auth status` command that:

1. Checks for a valid auth token in appdata (and make an API request to confirm its validity)
2. Prints a message to let the user know if they're authenticated, and, if so, what their username is

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Run `npm run cli:build`
2. Run `node dist/cli/main.js auth logout`
3. Run `node dist/cli/main.js auth status`
4. Ensure a "Authentication token is invalid or expired" message is printed
5. Open Studio
6. Log in to WordPress.com
7. Run `node dist/cli/main.js auth status`
8. Ensure a "Successfully authenticated with WordPress.com as \`USERNAME\`" message is printed

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Have you checked for TypeScript, React or other console errors?

fredrikekelund committed:
CLI: Implement `auth login` command

fredrikekelund committed:
Fix lint and tests

fredrikekelund committed:
Add tests, tweak command

fredrikekelund committed:
Consider expiration time

fredrikekelund committed:
Move logic around

fredrikekelund committed:
Unwrap

fredrikekelund committed:
Fix type error

fredrikekelund committed:
CLI: Implement `auth logout` command

fredrikekelund committed:
CLI: Implement `auth status` command

fredrikekelund committed:
Address review comments

fredrikekelund committed:
Reject unsupported platforms

fredrikekelund committed:
Merge branch 'f26d/cli-auth-login-command' into f26d/cli-auth-logout-command

fredrikekelund committed:
Merge branch 'f26d/cli-auth-logout-command' into f26d/cli-auth-status-command

youknowriad commented on cli/lib/api.ts:
There's a lot of things in this file that should be "common" between CLI and App.

fredrikekelund committed:
Merge branch 'trunk' into f26d/cli-auth-login-command

fredrikekelund committed:
Address review feedback

fredrikekelund committed:
Fix test

fredrikekelund committed:
Merge branch 'f26d/cli-auth-login-command' into f26d/cli-auth-logout-command

fredrikekelund committed:
Fix

fredrikekelund committed:
Merge branch 'f26d/cli-auth-logout-command' into f26d/cli-auth-status-command

fredrikekelund committed:
Merge branch 'trunk' into f26d/cli-auth-login-command

github-actions[bot] committed:
docs: Update CLI documentation for auth login command

Add comprehensive documentation for the new `studio auth login` command
including usage instructions, flow description, and important notes about
token lifetime and authentication sharing between CLI and desktop app.

Co-authored-by: Fredrik Rombach Ekelund <undefined@users.noreply.github.com>

fredrikekelund committed:
Restore package-lock.json

fredrikekelund committed:
Install inquirer dependency again

fredrikekelund committed:
Tweak

fredrikekelund committed:
Merge branch 'f26d/cli-auth-login-command' into f26d/cli-auth-logout-command

fredrikekelund committed:
Merge branch 'f26d/cli-auth-logout-command' into f26d/cli-auth-status-command

github-actions[bot] committed:
docs: Update CLI documentation for auth commands

Add documentation for `auth logout` and `auth status` commands.

Co-authored-by: Fredrik Rombach Ekelund <undefined@users.noreply.github.com>

bcotrim reviewed (approved):
LGTM! 👍 

Tested and works as described on macOS and Windows

fredrikekelund committed:
Merge branch 'trunk' into f26d/cli-auth-status-command

github-actions[bot] committed:
docs: Add CLI documentation for auth status command

Co-authored-by: Fredrik Rombach Ekelund <undefined@users.noreply.github.com>
